src/TileManager.js

```diff
@@ -173,32 +183,35 @@ export function getTileUrls(layer, bounds, zoom) {
  *
  * @return {object} geojson
  */
-export function getStoredTilesAsJson(layer, tiles) {
+export function getStoredTilesAsJson(layer, tiles, crs = L.CRS.EPSG3857) {
   const featureCollection = {
     type: 'FeatureCollection',
     features: [],
   };
+  const tileSize = layer.getTileSize();
   for (let i = 0; i < tiles.length; i += 1) {
-    const topLeftPoint = new L.Point(
-      tiles[i].x * layer.getTileSize().x,
-      tiles[i].y * layer.getTileSize().y,
+    const tile = tiles[i];
+    let { y } = tile;
+    const { x, z: zoom } = tile;
+    const worldTileBounds = layer._pxBoundsToTileRange(
+      crs.getProjectedBounds(zoom),
     );
+    if (layer.options.tms) {
+      const invertedY = worldTileBounds.max.y - y;
```

Comment:
The calculation of invertedY seems to share some code with the getTileUrls method. Is it possible to combine this part into one function?